ibis/expr/operations.py

```diff
@@ -2403,9 +2407,9 @@ class ArraySlice(ValueOp):
     input_type = [
         rules.array(dt.any),
         rules.integer(name='start'),
-        rules.integer(name='stop')
+        rules.integer(name='stop', optional=True)
```

Comment:
The downside to this approach is that you don't get a failure at expression creation time, instead you get a failure at compile time.